mlrun/runtimes/sparkclientjob.py

```diff
@@ -0,0 +1,141 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import re
+
+from .pod import KubeResourceSpec
+from mlrun.runtimes import KubejobRuntime, KubeRuntimeHandler
+from mlrun.config import config
+from mlrun.db import get_run_db
+from ..execution import MLClientCtx
+from ..model import RunObject
+from ..platforms.iguazio import mount_v3io_extended, mount_v3iod
+from subprocess import run
+
+
+class SparkClientSpec(KubeResourceSpec):
+    def __init__(
+        self,
+        command=None,
+        args=None,
+        image=None,
+        mode=None,
+        volumes=None,
+        volume_mounts=None,
+        env=None,
+        resources=None,
+        default_handler=None,
+        entry_points=None,
+        description=None,
+        workdir=None,
+        replicas=None,
+        image_pull_policy=None,
+        service_account=None,
+        build=None,
+        image_pull_secret=None,
+        igz_spark=None,
+    ):
+        super().__init__(
+            command=command,
+            args=args,
+            image=image,
+            mode=mode,
+            build=build,
+            entry_points=entry_points,
+            description=description,
+            workdir=workdir,
+            default_handler=default_handler,
+        )
+        self.igz_spark = igz_spark
+
+
+class SparkClientRuntime(KubejobRuntime):
+    kind = "sparkclient"
+
+    @property
+    def spec(self) -> SparkClientSpec:
+        return self._spec
+
+    @spec.setter
+    def spec(self, spec):
+        self._spec = self._verify_dict(spec, "spec", SparkClientSpec)
+
+    def with_igz_spark(self, spark_service):
+        self.spec.igz_spark = True
+        self.spec.env.append({"name": "MLRUN_SPARK_CLIENT_IGZ_SPARK", "value": "true"})
+        self.apply(mount_v3io_extended())
+        self.apply(
+            mount_v3iod(
+                namespace="default-tenant",
+                v3io_config_configmap=spark_service + "-submit",
+            )
+        )
+
+    @property
+    def is_deployed(self):
+        if (
+            not self.spec.build.source
+            and not self.spec.build.commands
+            and not self.spec.build.extra
+            and self.spec.igz_spark
+        ):
+            return True
+        return super().is_deployed
+
+    @property
+    def _default_image(self):
+        if (
+            self.spec.igz_spark
+            and config.spark_app_image
+            and config.spark_app_image_tag
+        ):
+            app_image = re.sub("spark-app", "shell", config.spark_app_image)
+            # this is temporary until we get the image name from external config
+            return app_image + ":" + config.spark_app_image_tag
+        return None
+
+    def deploy(self, watch=True, with_mlrun=True, skip_deployed=False, is_kfp=False):
+        """deploy function, build container with dependencies"""
+        # connect will populate the config from the server config
+        get_run_db()
```

Comment:
Not needed anymore, remove

Reply:
@Hedingber Does it mean that config will be filled on the client side as soon as we use new_function?

Reply:
Does it mean that it won't be needed in with_igz_spark too? (In order to get the namespace)

Reply:
It's done as soon as you `import mlrun` or setting the `config.dbpath` so yeah not needed anywhere